webapp/tests/test_functions.py

```diff
@@ -20,6 +20,439 @@ def return_less(series, value):
+
+    def test_safeMax_mixed(self):
+        self.assertEqual(functions.safeMax([10,None,5,None]), 10)
+
+
+    #
+    # Test safeAbs()
+    #
+
+    def test_safeAbs_None(self):
+        self.assertEqual(functions.safeAbs(None), None)
+
+    def test_safeAbs_empty_list(self):
+        with self.assertRaises(TypeError):
+          functions.safeAbs([])
+
+    def test_safeAbs_pos_number(self):
+        self.assertEqual(functions.safeAbs(1), 1)
+
+    def test_safeAbs_neg_numbers(self):
+        self.assertEqual(functions.safeAbs(-1), 1)
+
+    def test_safeAbs_zero(self):
+        self.assertEqual(functions.safeAbs(0), 0)
+
+    #
+    # Test safeMap()
+    #
+
+    def test_safeMap_None(self):
+        with self.assertRaises(TypeError):
+            functions.safeMap(abs, None)
+
+    def test_safeMap_empty_list(self):
+        self.assertEqual(functions.safeMap(abs, []), None)
+
+    def test_safeMap_all_numbers(self):
+        self.assertEqual(functions.safeMap(abs, [1,2,3,4]), [1,2,3,4])
+
+    def test_safeMap_all_None(self):
+        self.assertEqual(functions.safeMap(abs, [None,None,None,None]), None)
+
+    def test_safeMap_mixed(self):
+        self.assertEqual(functions.safeMap(abs, [10,None,5,None]), [10,5])
+
+    #
+    # Test gcd()
+    #
+
+    def test_gcd_None_None(self):
+        with self.assertRaises(TypeError):
+            functions.gcd(None, None)
+
+    def test_gcd_5_None(self):
+        with self.assertRaises(TypeError):
+            functions.gcd(5, None)
+
+    def test_gcd_5_0(self):
+        self.assertEqual(functions.gcd(5, 0), 5)
+
+    def test_gcd_0_10(self):
+        self.assertEqual(functions.gcd(0,10), 10)
+
+    def test_gcd_10_5(self):
+        self.assertEqual(functions.gcd(10,5), 5)
+
+    #
+    # Test lcm()
+    #
+
+    def test_lcm_None_None(self):
+        self.assertEqual(functions.lcm(None, None), None)
+
+    def test_lcm_5_None(self):
+        with self.assertRaises(TypeError):
+            functions.lcm(5, None)
+
+    def test_lcm_5_0(self):
+        self.assertEqual(functions.lcm(5, 0), 0)
+
+    def test_lcm_0_10(self):
+        self.assertEqual(functions.lcm(0,10), 0)
+
+    def test_lcm_10_5(self):
+        self.assertEqual(functions.lcm(10,5), 10)
+
+    #
+    # Test normalize()
+    #
+
+    def test_normalize_empty(self):
+        try:
+            functions.normalize([])
+        except NormalizeEmptyResultError:
+            pass
+
+    def test_normalize_None_values(self):
+        seriesList = []
+        seriesList.append(TimeSeries("collectd.test-db{0}.load.value", 0, 5, 1, [None, None, None, None, None]))
+        self.assertEqual(functions.normalize([seriesList]), (seriesList, 0, 5, 1))
+
+    def test_normalize_generate_series_list_input(self):
+        seriesList = self._generate_series_list()
+        self.assertEqual(functions.normalize([seriesList]), (seriesList, 0, 101, 1))
+
+    #
+    # Test matchSeries()
+    #
+#def matchSeries(seriesList1, seriesList2):
```

Comment:
It wasn't intended to be, but that's definitely not done.  I'll clean it up tomorrow during my train ride.  Sorry for the extra garbage.